vim/merlin/autoload/merlin.py

```diff
@@ -399,16 +400,18 @@ def vim_loclist(vimvar, ignore_warnings):
     bufnr = vim.current.buffer.number
     nr = 0
     for error in errors:
-        ty = 'E'
-        if error['type'] == 'warning':
-            if vim.eval(ignore_warnings) == 'true':
-                continue
-            ty = 'W'
+        ty = 'W' if error['type'] == 'warning' else 'E'
         msg = re.sub(re_wspaces, " ", error['message']).replace("'", "''")
         if msg.startswith("Warning "):
             msg = msg[8:]
         elif msg.startswith("Error: "):
             msg = msg[7:]
+        elif msg.startswith("Error (warning"):
+            msg = re.sub(re_error_warning, r"\1: ", msg)
+            ty = 'E'
```

Comment:
Is that line really necessary?
If it is, shouldn't it instead be fixed in the backend (to make sure the `type` field is set properly)?

Reply:
The stripping is in line with the stripping of the error and warning prefixes in the lines above (to keep the loclist concise).

Do you think the type of these warnings should be changed to an error in the backend? If so, I'd gladly include that change in this PR.

Reply:
I was only talking about the `ty = 'E'` line.

> Do you think the type of these warnings should be changed to an error in the backend? If so, I'd gladly include that change in this PR.

Yes.